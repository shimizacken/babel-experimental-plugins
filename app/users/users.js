export const getUserZip = (user) => user.address.zip;