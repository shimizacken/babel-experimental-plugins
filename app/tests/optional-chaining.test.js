import { getUserZip } from "../users/users";

describe('optional chaining test suite', () => {
  
    const users = [{
        name: "Jimi",
        address: {
            zip: 1628546
        }
    },
    {
        name: "Lea"
    },
    {
        name: "Ted",
        address: {}
    }];
  
    it('should return undefined when nested object is undefined', () => {
      expect(getUserZip(users[0])).toMatchSnapshot();
  });
});
