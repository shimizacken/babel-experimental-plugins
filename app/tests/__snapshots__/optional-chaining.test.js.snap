// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optional chaining test suite should return undefined when nested object is undefined 1`] = `1628546`;
